Added 'lenient ops' module

bluenote10 opened the pull request:
@Araq By chance I saw your idea with lenient ops on IRC. I had the same idea before and I was often using stuff like this for numeric-expression-heavy stuff. Would be nice to have this in the stdlib.

I'm only not sure how to name the module. Just `lenient` is maybe not meaningful enough. Alternatives that came to my mind: `auto_ops`, `convert_ops`, `float_int_ops` -- nothing really catchy.

Araq commented on lib/pure/lenientops.nim:
```diff
@@ -0,0 +1,142 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2015 Andreas Rumpf, Dominik Picheta
```
Either use your own name here or "Nim contributors"

Araq commented on lib/pure/lenientops.nim:
```diff
@@ -0,0 +1,142 @@
+## module. Use with care.
+
+
+proc `+`[I: SomeInteger, F: SomeReal](i: I, f: F): F = F(i) + f
```
These should all be ``inline`` procs, better take no risks. :-)

Araq commented on lib/pure/lenientops.nim:
```diff
@@ -0,0 +1,160 @@
+proc `>`[I: SomeInteger, F: SomeReal](f: F, i: I): bool {.noSideEffect, inline.} =
+  f > F(i)
+
+when isMainModule:
```
Due to the way overloading and scoping works (inner module overloads are preferred in the case of an ambiguity) you need to move these tests into a different module.

bluenote10 replied:
No problem, will do. This will also reveal one of my favorite mistakes: forgetting the `*` ;)

bluenote10 commented:
I have finalized the PR a bit. Things to approve by reviewers:

- I have added warning for `==` operator. This is the only operator which really might cause some true harm, because the use cases of comparing at integer level or on float level are really both relevant. At first I though about removing the operator entirely. But this would look "incomplete" and I think soon people would open issues that `==` is missing here. That's why I decided to make it a warning instead. What do you think?
- I have added a link in the standard library overview, but I wasn't sure if I should place it under "core" or "math" or somewhere else. Where would you prefer it?

konqoro commented:
Dumb question: Should those use `toFloat` from system.nim?

Araq commented:
> What do you think?

Remove it and explain in the docs why it is missing.

> Where would you prefer it?

Core.
